code/modules/projectiles/projectile.dm

```diff
@@ -164,7 +164,7 @@
 			if(starting)
 				splatter_dir = get_dir(starting, target_loca)
 			var/obj/item/bodypart/B = L.get_bodypart(def_zone)
-			if(B.status == BODYPART_ROBOTIC) // So if you hit a robotic, it sparks instead of bloodspatters
+			if(istype(B) && B.status == BODYPART_ROBOTIC) // So if you hit a robotic, it sparks instead of bloodspatters

```

Comment:
```suggestion
			if(B && B.status == BODYPART_ROBOTIC) // So if you hit a robotic, it sparks instead of bloodspatters
```
In no case `get_bodypart()` should return an atom reference that is not an actual (or sort of, like chainsaw arm or synthetic armblade) bodypart.